software/contrib/master_clock.py

```diff
@@ -96,24 +110,25 @@ def __init__(self):
         self.calcSleepTime()
         self.getPulseWidth()
 
-        # Get asyncio event loop object
-        #self.el = asyncio.get_event_loop()
-
         self.tasks = []
         for n in range(6):
-            self.tasks.append(0)
+            self.tasks.append(0)            
 
         # Starts/Stops the master clock
         @b1.handler_falling
         def StartStop():
-            self.running = not self.running
+            if ticks_diff(ticks_ms(), b1.last_pressed()) > 500 and ticks_diff(ticks_ms(), b1.last_pressed()) < 4000:
```

Comment:
[optional] This could be simplified to `if 500 < ticks_diff(ticks_ms(), b1.last_pressed()) < 4000:` 
It's simpler but also saves resources since you are comparing times only once.

Reply:
I had no idea that Python could do comparisons like that. However, I would prefer to keep the original as it is more readable. Also, the < 4000 can be removed if they phantom button press issue is ever resolved: https://github.com/Allen-Synthesis/EuroPi/issues/132